BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/mappers/FileMapper.java

```diff
@@ -24,11 +31,42 @@ public FileDto map(File file, String path) {
     }
 
     public FileDto map(File file, String basePath, String requestPath) {
-        return map(file, getFolderPath(basePath, requestPath) + PATH_SEPARATOR +  file.getName());
+        return map(file, basePath, requestPath, false, "");
+    }
+
+    public FileDto map(File file, String basePath, String requestPath, boolean showArchive, String zipPath) {
```

Comment:
it looks hard for future usage to identify what this parameters are and which overload method need to be used. I understant that convert from a file into a dto , and that if showArchive is true zip file will be included. other than that is not that easy to identify by method signature what this do or what parameters are. 

so I think  use some names with helps a i.e

```
mapFile(Fiile file, String pathPrefix, boolean showStructure)
```

```
mapZipFile(Fiile zip, String fileInZip, String pathPrefix, boolean showStructure)
```